// @sentdm/sentdm/resources/profiles/campaigns.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Campaigns = void 0;
const resource_1 = require("../../core/resource.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Manage organization profiles
 */
class Campaigns extends resource_1.APIResource {
    /**
     * Creates a new campaign scoped under the brand of the specified profile. Each
     * campaign must include at least one use case with sample messages.
     *
     * @example
     * ```ts
     * const apiResponseOfTcrCampaignWithUseCases =
     *   await client.profiles.campaigns.create(
     *     '770e8400-e29b-41d4-a716-446655440002',
     *     {
     *       campaign: {
     *         description:
     *           'Appointment reminders and account notifications',
     *         name: 'Customer Notifications',
     *         type: 'App',
     *         useCases: [
     *           {
     *             messagingUseCaseUs: 'ACCOUNT_NOTIFICATION',
     *             sampleMessages: [
     *               'Hi {name}, your appointment is confirmed for {date} at {time}.',
     *               'Your order #{order_id} has been shipped. Track at {url}',
     *             ],
     *           },
     *         ],
     *       },
     *     },
     *   );
     * ```
     */
    create(profileID, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.post((0, path_1.path) `/v3/profiles/${profileID}/campaigns`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Updates an existing campaign under the brand of the specified profile. Cannot
     * update campaigns that have already been submitted to TCR.
     *
     * @example
     * ```ts
     * const apiResponseOfTcrCampaignWithUseCases =
     *   await client.profiles.campaigns.update(
     *     'b2c3d4e5-f6a7-8901-bcde-f12345678901',
     *     {
     *       profileId: '770e8400-e29b-41d4-a716-446655440002',
     *       campaign: {
     *         description:
     *           'Updated appointment reminders and account notifications',
     *         name: 'Customer Notifications Updated',
     *         type: 'App',
     *         useCases: [
     *           {
     *             messagingUseCaseUs: 'ACCOUNT_NOTIFICATION',
     *             sampleMessages: [
     *               'Hi {name}, your appointment is confirmed for {date} at {time}.',
     *               'Your order #{order_id} has been shipped. Track at {url}',
     *             ],
     *           },
     *         ],
     *       },
     *     },
     *   );
     * ```
     */
    update(campaignID, params, options) {
        const { profileId, 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.put((0, path_1.path) `/v3/profiles/${profileId}/campaigns/${campaignID}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves all campaigns linked to the profile's brand, including use cases and
     * sample messages. Returns inherited campaigns if inherit_tcr_campaign=true.
     *
     * @example
     * ```ts
     * const campaigns = await client.profiles.campaigns.list(
     *   '770e8400-e29b-41d4-a716-446655440002',
     * );
     * ```
     */
    list(profileID, params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.get((0, path_1.path) `/v3/profiles/${profileID}/campaigns`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Deletes a campaign by ID from the brand of the specified profile. The profile
     * must belong to the authenticated organization.
     *
     * @example
     * ```ts
     * await client.profiles.campaigns.delete(
     *   'b2c3d4e5-f6a7-8901-bcde-f12345678901',
     *   { profileId: '770e8400-e29b-41d4-a716-446655440002' },
     * );
     * ```
     */
    delete(campaignID, params, options) {
        const { profileId, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.delete((0, path_1.path) `/v3/profiles/${profileId}/campaigns/${campaignID}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { Accept: '*/*', ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
}
exports.Campaigns = Campaigns;
