// @sentdm/sentdm/resources/profiles/profiles.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as CampaignsAPI from "./campaigns.mjs";
import { Campaigns, } from "./campaigns.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Manage organization profiles
 */
export class Profiles extends APIResource {
    constructor() {
        super(...arguments);
        this.campaigns = new CampaignsAPI.Campaigns(this._client);
    }
    /**
     * Creates a new sender profile within an organization. Profiles represent
     * different brands, departments, or use cases, each with their own messaging
     * configuration and settings. Requires admin role in the organization.
     *
     * ## WhatsApp Business Account
     *
     * Every profile must be linked to a WhatsApp Business Account. There are two ways
     * to do this:
     *
     * **1. Inherit from organization (default)** — Omit the
     * `whatsapp_business_account` field. The profile will share the organization's
     * WhatsApp Business Account, which must have been set up via WhatsApp Embedded
     * Signup. This is the recommended path for most use cases.
     *
     * **2. Direct credentials** — Provide a `whatsapp_business_account` object with
     * `waba_id`, `phone_number_id`, and `access_token`. Use this when the profile
     * needs its own independent WhatsApp Business Account. Obtain these from Meta
     * Business Manager by creating a System User with `whatsapp_business_messaging`
     * and `whatsapp_business_management` permissions.
     *
     * If the `whatsapp_business_account` field is omitted and the organization has no
     * WhatsApp Business Account configured, the request will be rejected with
     * HTTP 422.
     *
     * ## Brand
     *
     * Include the optional `brand` field to create the brand for this profile at the
     * same time. Cannot be used when `inherit_tcr_brand` is `true`.
     *
     * ## Payment Details
     *
     * When `billing_model` is `"profile"` or `"profile_and_organization"` you may
     * include a `payment_details` object containing the card number, expiry (MM/YY),
     * CVC, and billing ZIP code. Payment details are **never stored** on our servers
     * and are forwarded directly to the payment processor. Providing `payment_details`
     * when `billing_model` is `"organization"` is not allowed.
     *
     * @example
     * ```ts
     * const apiResponseOfProfileDetail =
     *   await client.profiles.create();
     * ```
     */
    create(params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.post('/v3/profiles', {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves detailed information about a specific sender profile within an
     * organization, including brand and KYC information if a brand has been
     * configured.
     *
     * @example
     * ```ts
     * const apiResponseOfProfileDetail =
     *   await client.profiles.retrieve('profileId');
     * ```
     */
    retrieve(profileID, params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.get(path `/v3/profiles/${profileID}`, {
            ...options,
            headers: buildHeaders([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Updates a profile's configuration and settings. Requires admin role in the
     * organization. Only provided fields will be updated (partial update).
     *
     * ## Brand Management
     *
     * Include the optional `brand` field to create or update the brand associated with
     * this profile. The brand holds KYC and TCR compliance data (legal business info,
     * contact details, messaging vertical). Once a brand has been submitted to TCR it
     * cannot be modified. Setting `inherit_tcr_brand: true` and providing `brand` in
     * the same request is not allowed.
     *
     * ## Payment Details
     *
     * When `billing_model` is `"profile"` or `"profile_and_organization"` you may
     * include a `payment_details` object containing the card number, expiry (MM/YY),
     * CVC, and billing ZIP code. Payment details are **never stored** on our servers
     * and are forwarded directly to the payment processor. Providing `payment_details`
     * when `billing_model` is `"organization"` is not allowed.
     *
     * @example
     * ```ts
     * const apiResponseOfProfileDetail =
     *   await client.profiles.update('profileId');
     * ```
     */
    update(profileID, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.patch(path `/v3/profiles/${profileID}`, {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves all sender profiles within an organization, including brand
     * information for each profile. Profiles represent different brands, departments,
     * or use cases within an organization, each with their own messaging
     * configuration.
     *
     * @example
     * ```ts
     * const profiles = await client.profiles.list();
     * ```
     */
    list(params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.get('/v3/profiles', {
            ...options,
            headers: buildHeaders([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Soft deletes a sender profile. The profile will be marked as deleted but data is
     * retained. Requires admin role in the organization.
     *
     * @example
     * ```ts
     * await client.profiles.delete('profileId');
     * ```
     */
    delete(profileID, params, options) {
        const { 'x-profile-id': xProfileID, ...body } = params;
        return this._client.delete(path `/v3/profiles/${profileID}`, {
            body,
            ...options,
            headers: buildHeaders([
                { Accept: '*/*', ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Final step in profile compliance workflow. Validates all prerequisites (general
     * data, brand, campaigns), connects profile to Telnyx/WhatsApp, and sets status
     * based on configuration. The process runs in the background and calls the
     * provided webhook URL when finished.
     *
     *                 Prerequisites:
     *                 - Profile must be completed
     *                 - If inheritTcrBrand=false: Profile must have existing brand
     *                 - If inheritTcrBrand=true: Parent must have existing brand
     *                 - If TCR application: Must have at least one campaign (own or inherited)
     *                 - If inheritTcrCampaign=false: Profile should have campaigns
     *                 - If inheritTcrCampaign=true: Parent must have campaigns
     *
     *                 Status Logic:
     *                 - If both SMS and WhatsApp channels are missing → SUBMITTED
     *                 - If TCR application and not inheriting brand/campaigns → SUBMITTED
     *                 - If non-TCR with destination country (IsMain=true) → SUBMITTED
     *                 - Otherwise → COMPLETED
     *
     * @example
     * ```ts
     * const response = await client.profiles.complete(
     *   '660e8400-e29b-41d4-a716-446655440000',
     *   {
     *     webHookUrl:
     *       'https://your-app.com/webhook/profile-complete',
     *   },
     * );
     * ```
     */
    complete(profileID, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.post(path `/v3/profiles/${profileID}/complete`, {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
}
Profiles.Campaigns = Campaigns;
