// @sentdm/sentdm/resources/contacts.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Create, update, and manage customer contact lists
 */
export class Contacts extends APIResource {
    /**
     * Creates a new contact by phone number and associates it with the authenticated
     * customer.
     *
     * @example
     * ```ts
     * const apiResponseOfContact = await client.contacts.create();
     * ```
     */
    create(params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.post('/v3/contacts', {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves a specific contact by their unique identifier. Returns detailed
     * contact information including phone formats, available channels, and opt-out
     * status.
     *
     * @example
     * ```ts
     * const apiResponseOfContact = await client.contacts.retrieve(
     *   '6ba7b810-9dad-11d1-80b4-00c04fd430c8',
     * );
     * ```
     */
    retrieve(id, params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.get(path `/v3/contacts/${id}`, {
            ...options,
            headers: buildHeaders([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Updates a contact's default channel and/or opt-out status. Inherited contacts
     * cannot be updated.
     *
     * @example
     * ```ts
     * const apiResponseOfContact = await client.contacts.update(
     *   '6ba7b810-9dad-11d1-80b4-00c04fd430c8',
     * );
     * ```
     */
    update(id, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.patch(path `/v3/contacts/${id}`, {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves a paginated list of contacts for the authenticated customer. Supports
     * filtering by search term, channel, or phone number.
     *
     * @example
     * ```ts
     * const contacts = await client.contacts.list({
     *   page: 0,
     *   page_size: 0,
     * });
     * ```
     */
    list(params, options) {
        const { 'x-profile-id': xProfileID, ...query } = params;
        return this._client.get('/v3/contacts', {
            query,
            ...options,
            headers: buildHeaders([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Dissociates a contact from the authenticated customer. Inherited contacts cannot
     * be deleted.
     *
     * @example
     * ```ts
     * await client.contacts.delete(
     *   '6ba7b810-9dad-11d1-80b4-00c04fd430c8',
     * );
     * ```
     */
    delete(id, params, options) {
        const { 'x-profile-id': xProfileID, ...body } = params;
        return this._client.delete(path `/v3/contacts/${id}`, {
            body,
            ...options,
            headers: buildHeaders([
                { Accept: '*/*', ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
}
