// @sentdm/sentdm/resources/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhooks = exports.Users = exports.Templates = exports.Profiles = exports.Numbers = exports.Messages = exports.Me = exports.Contacts = void 0;
var contacts_1 = require("./contacts.js");
Object.defineProperty(exports, "Contacts", { enumerable: true, get: function () { return contacts_1.Contacts; } });
var me_1 = require("./me.js");
Object.defineProperty(exports, "Me", { enumerable: true, get: function () { return me_1.Me; } });
var messages_1 = require("./messages.js");
Object.defineProperty(exports, "Messages", { enumerable: true, get: function () { return messages_1.Messages; } });
var numbers_1 = require("./numbers.js");
Object.defineProperty(exports, "Numbers", { enumerable: true, get: function () { return numbers_1.Numbers; } });
var profiles_1 = require("./profiles/profiles.js");
Object.defineProperty(exports, "Profiles", { enumerable: true, get: function () { return profiles_1.Profiles; } });
var templates_1 = require("./templates.js");
Object.defineProperty(exports, "Templates", { enumerable: true, get: function () { return templates_1.Templates; } });
var users_1 = require("./users.js");
Object.defineProperty(exports, "Users", { enumerable: true, get: function () { return users_1.Users; } });
var webhooks_1 = require("./webhooks.js");
Object.defineProperty(exports, "Webhooks", { enumerable: true, get: function () { return webhooks_1.Webhooks; } });
