// @sentdm/sentdm/resources/templates.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Templates = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Manage message templates with variable substitution
 */
class Templates extends resource_1.APIResource {
    /**
     * Creates a new message template with header, body, footer, and buttons. The
     * template can be submitted for review immediately or saved as draft for later
     * submission.
     *
     * @example
     * ```ts
     * const apiResponseTemplate = await client.templates.create();
     * ```
     */
    create(params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.post('/v3/templates', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves a specific template by its ID. Returns template details including
     * name, category, language, status, and definition.
     *
     * @example
     * ```ts
     * const apiResponseTemplate = await client.templates.retrieve(
     *   '7ba7b820-9dad-11d1-80b4-00c04fd430c8',
     * );
     * ```
     */
    retrieve(id, params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.get((0, path_1.path) `/v3/templates/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Updates an existing template's name, category, language, definition, or submits
     * it for review.
     *
     * @example
     * ```ts
     * const apiResponseTemplate = await client.templates.update(
     *   '7ba7b820-9dad-11d1-80b4-00c04fd430c8',
     * );
     * ```
     */
    update(id, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.put((0, path_1.path) `/v3/templates/${id}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves a paginated list of message templates for the authenticated customer.
     * Supports filtering by status, category, and search term.
     *
     * @example
     * ```ts
     * const templates = await client.templates.list({
     *   page: 0,
     *   page_size: 0,
     * });
     * ```
     */
    list(params, options) {
        const { 'x-profile-id': xProfileID, ...query } = params;
        return this._client.get('/v3/templates', {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Deletes a template by ID. Optionally, you can also delete the template from
     * WhatsApp/Meta by setting delete_from_meta=true.
     *
     * @example
     * ```ts
     * await client.templates.delete(
     *   '7ba7b820-9dad-11d1-80b4-00c04fd430c8',
     * );
     * ```
     */
    delete(id, params, options) {
        const { 'x-profile-id': xProfileID, ...body } = params;
        return this._client.delete((0, path_1.path) `/v3/templates/${id}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { Accept: '*/*', ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
}
exports.Templates = Templates;
