// @sentdm/sentdm/resources/users.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
/**
 * Invite, update, and manage organization users and roles
 */
export class Users extends APIResource {
    /**
     * Retrieves detailed information about a specific user in an organization or
     * profile. Requires developer role or higher.
     *
     * @example
     * ```ts
     * const apiResponseOfUser = await client.users.retrieve(
     *   'userId',
     * );
     * ```
     */
    retrieve(userID, params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.get(path `/v3/users/${userID}`, {
            ...options,
            headers: buildHeaders([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves all users who have access to the organization or profile identified by
     * the API key, including their roles and status. Shows invited users (pending
     * acceptance) and active users. Requires developer role or higher.
     *
     * @example
     * ```ts
     * const users = await client.users.list();
     * ```
     */
    list(params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.get('/v3/users', {
            ...options,
            headers: buildHeaders([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Sends an invitation to a user to join the organization or profile with a
     * specific role. Requires admin role. The user will receive an invitation email
     * with a token to accept. Invitation tokens expire after 7 days.
     *
     * @example
     * ```ts
     * const apiResponseOfUser = await client.users.invite();
     * ```
     */
    invite(params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.post('/v3/users', {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Removes a user's access to an organization or profile. Requires admin role. You
     * cannot remove yourself or remove the last admin.
     *
     * @example
     * ```ts
     * await client.users.remove('userId');
     * ```
     */
    remove(userID, params, options) {
        const { 'x-profile-id': xProfileID, ...body } = params;
        return this._client.delete(path `/v3/users/${userID}`, {
            body,
            ...options,
            headers: buildHeaders([
                { Accept: '*/*', ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Updates a user's role in the organization or profile. Requires admin role. You
     * cannot change your own role or demote the last admin.
     *
     * @example
     * ```ts
     * const apiResponseOfUser = await client.users.updateRole(
     *   'userId',
     * );
     * ```
     */
    updateRole(userID, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.patch(path `/v3/users/${userID}`, {
            body,
            ...options,
            headers: buildHeaders([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
}
