// @sentdm/sentdm/resources/webhooks.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhooks = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
/**
 * Configure webhook endpoints for real-time event delivery
 */
class Webhooks extends resource_1.APIResource {
    /**
     * Creates a new webhook endpoint for the authenticated customer.
     *
     * @example
     * ```ts
     * const apiResponseWebhook = await client.webhooks.create();
     * ```
     */
    create(params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.post('/v3/webhooks', {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves a single webhook by ID for the authenticated customer.
     *
     * @example
     * ```ts
     * const apiResponseWebhook = await client.webhooks.retrieve(
     *   'd4f5a6b7-c8d9-4e0f-a1b2-c3d4e5f6a7b8',
     * );
     * ```
     */
    retrieve(id, params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.get((0, path_1.path) `/v3/webhooks/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Updates an existing webhook for the authenticated customer.
     *
     * @example
     * ```ts
     * const apiResponseWebhook = await client.webhooks.update(
     *   'd4f5a6b7-c8d9-4e0f-a1b2-c3d4e5f6a7b8',
     * );
     * ```
     */
    update(id, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.put((0, path_1.path) `/v3/webhooks/${id}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves a paginated list of webhooks for the authenticated customer.
     *
     * @example
     * ```ts
     * const webhooks = await client.webhooks.list({
     *   page: 0,
     *   page_size: 0,
     * });
     * ```
     */
    list(params, options) {
        const { 'x-profile-id': xProfileID, ...query } = params;
        return this._client.get('/v3/webhooks', {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Deletes a webhook for the authenticated customer.
     *
     * @example
     * ```ts
     * await client.webhooks.delete(
     *   'd4f5a6b7-c8d9-4e0f-a1b2-c3d4e5f6a7b8',
     * );
     * ```
     */
    delete(id, params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.delete((0, path_1.path) `/v3/webhooks/${id}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { Accept: '*/*', ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves all available webhook event types that can be subscribed to.
     *
     * @example
     * ```ts
     * const response = await client.webhooks.listEventTypes();
     * ```
     */
    listEventTypes(params = {}, options) {
        const { 'x-profile-id': xProfileID } = params ?? {};
        return this._client.get('/v3/webhooks/event-types', {
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Retrieves a paginated list of delivery events for the specified webhook.
     *
     * @example
     * ```ts
     * const response = await client.webhooks.listEvents(
     *   'd4f5a6b7-c8d9-4e0f-a1b2-c3d4e5f6a7b8',
     *   { page: 0, page_size: 0 },
     * );
     * ```
     */
    listEvents(id, params, options) {
        const { 'x-profile-id': xProfileID, ...query } = params;
        return this._client.get((0, path_1.path) `/v3/webhooks/${id}/events`, {
            query,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                { ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined) },
                options?.headers,
            ]),
        });
    }
    /**
     * Generates a new signing secret for the specified webhook. The old secret is
     * immediately invalidated.
     *
     * @example
     * ```ts
     * const response = await client.webhooks.rotateSecret(
     *   'd4f5a6b7-c8d9-4e0f-a1b2-c3d4e5f6a7b8',
     * );
     * ```
     */
    rotateSecret(id, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.post((0, path_1.path) `/v3/webhooks/${id}/rotate-secret`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Sends a test event to the specified webhook endpoint to verify connectivity.
     *
     * @example
     * ```ts
     * const response = await client.webhooks.test(
     *   'd4f5a6b7-c8d9-4e0f-a1b2-c3d4e5f6a7b8',
     * );
     * ```
     */
    test(id, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.post((0, path_1.path) `/v3/webhooks/${id}/test`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
    /**
     * Activates or deactivates a webhook for the authenticated customer.
     *
     * @example
     * ```ts
     * const apiResponseWebhook =
     *   await client.webhooks.toggleStatus(
     *     'd4f5a6b7-c8d9-4e0f-a1b2-c3d4e5f6a7b8',
     *   );
     * ```
     */
    toggleStatus(id, params, options) {
        const { 'Idempotency-Key': idempotencyKey, 'x-profile-id': xProfileID, ...body } = params;
        return this._client.patch((0, path_1.path) `/v3/webhooks/${id}/toggle-status`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([
                {
                    ...(idempotencyKey != null ? { 'Idempotency-Key': idempotencyKey } : undefined),
                    ...(xProfileID != null ? { 'x-profile-id': xProfileID } : undefined),
                },
                options?.headers,
            ]),
        });
    }
}
exports.Webhooks = Webhooks;
